torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3008,6 +3008,160 @@ struct to_ir {
         return std::make_shared<SimpleValue>(
             emitListComprehension(lc, type_hint));
       }
+      case prim::dict: {
+
+        auto treat_as_empty_container = [&]{
+          if (apply.inputs().empty() && !apply.attributes().empty()) {
+            return true;
+          }
+          if (!apply.inputs().empty() && apply.inputs()[0].kind() == TK_DICT_LITERAL) {
+            auto dict_lit = DictLiteral(apply.inputs()[0]);
+            return dict_lit.key_inputs().empty() && dict_lit.value_inputs().empty();
+          }
+          if (!apply.inputs().empty() && apply.inputs()[0].kind() == TK_LIST_LITERAL) {
+            auto list_lit = ListLiteral(apply.inputs()[0]);
+            return list_lit.inputs().empty();
+          }
+          return false;
+        };
+
+        // If we have a call with an empty container, or if we have a
+        // call with kwargs only
+        if (treat_as_empty_container()) {
```

Comment:
I think if we got rid of this case the `dict({})` case would work correctly and the empthy list case would error(which is fine)